### DISCUSSION TITLE
How to plot Rarefaction curve showing standard deviation in the curve and colored by groups or metadata

### DISCUSSION THREAD

**Discussion by DPC7** [vegandevs/vegan#480] [Q&A]
Hi I am using phyloseq and vegan to plot rarefaction graph. I only succeed to plot a curve like this: 
![rarefaction_bio](https://user-images.githubusercontent.com/61667261/88351505-2cd0c000-cd74-11ea-9e05-e628d21f2a44.png)
But, I want the x axis to be number of samples, and curves to be plotted on the basis of metadata information (e.g. sample_type (control, test), etc.). Exactly like the curve in [this article page ](https://drive.google.com/file/d/14t4otKDGeuyLEOYNGaGUNAnKssnZqPtM/view?usp=drivesdk) .

Can anyone please help me out?

Thanks
dpc

**Comment by jarioksa** [comment]
If you want to show the accumulation by sampling units, you should use species accumulation curves instead of rarefaction (rarefaction is within sampling units, species accumulation is among sampling units). Check function `specaccum` in **vegan**. The default `method = "exact"` is similar to the rarefaction method, and argument `subset` can be used to select a subset of sampling units. Two accumulations can be plotted by setting argument `add = TRUE` for the second `plot` command.

**Comment by DPC7** [comment]
Thanks a

> If you want to show the accumulation by sampling units, you should use species accumulation curves instead of rarefaction (rarefaction is within sampling units, species accumulation is among sampling units). Check function `specaccum` in **vegan**. The default `method = "exact"` is similar to the rarefaction method, and argument `subset` can be used to select a subset of sampling units. Two accumulations can be plotted by setting argument `add = TRUE` for the second `plot` command.

Thanks a lot @jarioksa for helping me out. I have followed this [tutorial](https://rdrr.io/rforge/vegan/man/specaccum.html) and able to generate graph as per the tutorial. BUT, how can I see t**wo different curves for control and test sample groups** in my metadata?

DC7

**Comment by jarioksa** [comment]
I think I told that in my message. Look at the last sentence.

**Comment by DPC7** [comment]
Thanks sir... yes, I have done it.... I didn't understand that at very first.... 

Thanks a lot for helping me out...

DC7

**Comment by DPC7** [comment]
> If you want to show the accumulation by sampling units, you should use species accumulation curves instead of rarefaction (rarefaction is within sampling units, species accumulation is among sampling units). Check function `specaccum` in **vegan**. The default `method = "exact"` is similar to the rarefaction method, and argument `subset` can be used to select a subset of sampling units. Two accumulations can be plotted by setting argument `add = TRUE` for the second `plot` command.

Sir @jarioksa  , then why they captioned as "rarefaction"? They did not get a curve for each of the samples.
